### ISSUE TITLE
See Vanished Check Wasted Cycles

### ISSUE THREAD

**Issue by cjm721** [CodeCrafter47/BungeeTabListPlus#293]
bungeetablistplus.seevanished is being pinged for every player about every second. This happening even when there are no vanished players.

Is there a way to disable this compatibility or short circuit the check so that if there are no vanished players it does not attempt to check if players have permission to see noone.

**Comment by CodeCrafter47** [comment]
Permission queries are cheap, so this should not really be an issue.

However if it bothers you, you can configure BungeeTabListPlus to always/ never display vanished players, which will remove the permission check.
Whether invisible players are to be included can be configured per player set. You can find information on how to do it on the following wiki page: <https://github.com/CodeCrafter47/BungeeTabListPlus/wiki/PlayerSets>. If you always explicitly set the `hiddenPlayers` option to either `VISIBLE` or `INVISIBLE` then the plugin won't be checking the `bungeetablistplus.seevanished` permission. Note that you may need to restart BungeeCord to see the effect.

**Comment by cjm721** [comment]
Changing that setting did not seem to have the desired effect. [Config](https://gist.github.com/cjm721/705bb6a4572ffceb55d2440ab001a81b)

From looking through the code real fast it looks like the check for permissions is done before it checks the playerSet's visibility settings so it can't short circuit on the quick boolean checks.

https://github.com/CodeCrafter47/BungeeTabListPlus/blob/0d431cd483da4c9323316e708c491a776c7ac861/bungee/src/main/java/codecrafter47/bungeetablistplus/tablistproviders/ConfigTablistProvider.java#L132

**Comment by CodeCrafter47** [comment]
Looks like you're right.

However I am a bit reluctant to changing the code as I see no clear benefits. Can you explain why you want to remove these permission checks? Because the way I see it, if this is about improving the performance of the plugin then there are definitely other places to look at first, on the other hand if these permission checks are expensive(take a long time) in your setup, then something is wrong with your permission plugin.

**Comment by cjm721** [comment]
I caught it when I was doing some debuging of another issue and turned off my cache system so every lookup was a hard call to the database, it becomes very annoying when the just the existence of something causes calls when it can already know the outcome based on its current state. 

In production it won't really be noticeable, but for a simple change of order of transitive operations it would make sense to do.

**Comment by CodeCrafter47** [comment]
Looking at the code again I think the permission check was purposefully done outside the loop to avoid having to do lots of permission checks in the case there are vanished players. So simple change of order won't  do it, but this can definitely be optimized. I'll look into it when I get around to optimizing player sets.

**Comment by andrewkm** [comment]
Would be really nice if you can optimize this @CodeCrafter47 this is the only plugin causing a  major amount of permissions spam.

**Comment by CodeCrafter47** [comment]
In an update in the far future the permission checks will go away if the `VISIBLE_TO_ADMINS` option is not used. If it is used, then the checks are necessary.

**Comment by andrewkm** [comment]
Any updates on this one @CodeCrafter47

**Comment by CodeCrafter47** [comment]
I'll fix it when it's convenient. This issue is low priority because it doesn't affect functionality of the plugin.

**Comment by andrewkm** [comment]
Still waiting on this, it's making our permissions debugging and verbose with LuckPerms a terrible mess.

**Comment by andrewkm** [comment]
@CodeCrafter47 just about ready to offer my left kidney for this. Let me know when to freeze it and send it over.

**Comment by CodeCrafter47** [comment]
The permission should no longer be checked if you are not using `VISIBLE_TO_ADMINS`. Though currently a reload may not be sufficient for the plugin to stop querying the permission, you may need a full restart after changing the config.
